### ISSUE TITLE
Belief-state cluster omits the pretrained-HMM behavioral and algorithm-extraction bridge

### ISSUE THREAD

**Issue by microprediction** [microprediction/latentreasoning#156]
## Reviewed source

- Default-branch commit: [`88d1db6b1c014f641f67ec4b84ec1f51e8f0c43b`](https://github.com/microprediction/latentreasoning/commit/88d1db6b1c014f641f67ec4b84ec1f51e8f0c43b)
- Affected section: [`bibliography.html`, “Belief state, evidence and attribution,” lines 1903–1941](https://github.com/microprediction/latentreasoning/blob/88d1db6b1c014f641f67ec4b84ec1f51e8f0c43b/bibliography.html#L1903-L1941)
- Current endpoint paper: Balcells et al., [*Large Language Models Develop Belief State Geometry In-Context*](https://arxiv.org/abs/2609.17376)
- Missing primary sources:
  - Dai et al., [*Pre-trained Large Language Models Learn to Predict Hidden Markov Models In-context*, arXiv:2506.07298v3 / NeurIPS 2025](https://arxiv.org/html/2506.07298v3) ([NeurIPS proceedings](https://papers.nips.cc/paper_files/paper/2025/hash/29a0ea49a103a233b17c0705cdeccb66-Abstract-Conference.html))
  - Dai et al., [*Extracting Algorithms in Pre-trained LLMs: A Case on Hidden Markov Models*, arXiv:2607.22646v1](https://arxiv.org/html/2607.22646)

## Material missing link

The section currently places Balcells et al. (2026), on prompted production-scale LLMs, directly next to Shai et al. (2024), the small-transformer trained-on-HMM antecedent. Two papers that bridge those regimes are absent.

This is not merely an extra related-work suggestion. Balcells et al. explicitly use Dai et al. (2025) as the behavioral foundation for the experiment:

- their introduction says open-source LLMs already predict HMM-generated sequences competently in context, citing Dai;
- their 40 HMMs are selected within the entropy/mixing-rate regime characterized by Dai;
- they motivate the activation study by combining Shai's trained-small-model geometry with Dai's pretrained-model behavior.

The omitted 2025 paper tests pretrained LLM next-observation prediction on synthetic HMMs across context length, entropy and mixing regimes, and also evaluates animal decision sequences. It establishes behavior, not a belief-state representation.

The omitted 2026 follow-up is even closer to this section's subject. It benchmarks 12 open-weight models on 75 HMM configurations against forward-filtering, spectral, linear/nonlinear n-gram and Baum–Welch predictors. It finds that no single candidate class matches behavior across all regimes. Its Principal Activations Probe then separates linear decodability from intervention effects and reports regime-dependent causal effects of candidate algorithmic representations. This predates the Balcells preprint and directly concerns internal representations in pretrained LLMs.

Without these papers, the cluster presents an incomplete progression:

```
trained toy transformer geometry (2024)
    -> [missing pretrained behavior (2025)]
    -> [missing candidate-algorithm interventions (June 2026)]
    -> prompted belief-geometry interventions (September 2026)
```

That omission matters because the papers establish different things: predictive behavior, candidate algorithmic mechanisms, and posterior-belief geometry should not be collapsed into one result.

## Important correction when adding the 2025 paper

The 2025 paper repeatedly calls its Viterbi baseline the “theoretical optimum.” That is not generally correct for next-observation prediction.

Viterbi selects one maximum-probability hidden-state **path** and the implementation predicts from its final state. The Bayes-optimal predictive distribution instead marginalizes over the filtered posterior:

```
P(O_(t+1) | O_(1:t))
  = sum_h P(H_t=h | O_(1:t))
          sum_h' A[h,h'] B[h', O_(t+1)].
```

A simple counterexample: suppose the filtered posterior over two current states is `(0.6, 0.4)`, transitions are identity, and `P(Y=1|A)=0.9`, `P(Y=1|B)=0.1`. A Viterbi-state prediction uses `0.9`; the Bayes predictive probability is `0.6*0.9 + 0.4*0.1 = 0.58`. Under log loss, Brier score, or any strictly proper score, `0.58` is optimal and `0.9` is not.

The 2026 follow-up fixes the benchmark definition: its Oracle uses the **forward recursion** and returns the Bayes-optimal next-observation distribution (§2.2, Table 1). Thus a new annotation should say the 2025 LLMs approach the paper's ground-truth-parameter Viterbi baseline on the tested accuracy metric, not an unconditional theoretical optimum.

## Suggested additions

For example, insert between Shai (2024) and Balcells (2026), or order the cluster chronologically:

> **Dai et al. (2025).** Pretrained LLMs predict synthetic HMM observation streams increasingly well with context, with performance depending on entropy and mixing rate, and are competitive with domain-specific models on two animal-decision datasets. The paper benchmarks against a ground-truth-parameter Viterbi-state predictor; despite its wording, that is not the Bayes-optimal next-observation distribution, which requires forward marginalization.

> **Dai et al. (2026).** Compares 12 pretrained LLMs with spectral, linear/nonlinear n-gram and Baum–Welch predictors on 75 HMM configurations; no single class explains all regimes. Principal-activation probes plus interventions find regime-dependent candidate-algorithm signals and show that decodability and causal use can dissociate. The oracle here correctly uses forward filtering.

This supplies the missing behavioral and mechanism bridge while preserving the distinct contribution of Balcells et al.: supervised, per-sequence belief-geometry probes and targeted belief interventions.

**Comment by microprediction** [comment]
Added both in ff85fa5. The 2025 paper (arXiv:2506.07298, §2.3) defines convergence against a Viterbi algorithm that 'computes the most likely hidden state sequence', which it calls the theoretical optimum. That isn't the Bayes predictive distribution, and the entry says so. The 2026 follow-up (arXiv:2607.22646) uses the forward recursion as its oracle, reports belief decodability dissociating from patching in early layers (§4.3), and finds soft n-gram the most causally effective representation. I didn't state that Balcells et al. build on Dai (2025), as I couldn't check it.
